UPGRADE-3.x.md

```diff
@@ -1,4 +1,10 @@
 UPGRADE 3.x
+
+## Deprecated passing no 3rd argument to GroupMenuProvider
+
+Passing no 3rd argument to `Menu\Provider\GroupMenuProvider` is deprecated.
+Pass `Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface` as 3rd argument.
```

Comment:
As I talked to @OskarStark, I don't think in NEXT_MAJOR `SessionContext` will be available becuase of the Symfony dependency. So, I think for a deprecation message we should not inform users to use an argument which won't be available.